### PR TITLE
Add element-wise multiplication for matrix

### PR DESCRIPTION
**Pull request by Suckzoo**
Hello,
I tried to find a function that multiplies matrices element-wise in your library, but it wasn't there.
So I implemented it.

### CODE REVIEW

File: src/abstractMatrix.js [hunk 1]
```diff
@@ -980,6 +980,33 @@ export default function AbstractMatrix(superCtor) {
...
+    /**
+         * Returns the element-wise product between this and other
+         * @param {Matrix} other
+         * @return {Matrix}
+         */
+    emul(other) {
+      other = this.constructor.checkMatrix(other);
+      if (this.rows !== other.rows || this.columns !== other.columns) {
+        // eslint-disable-next-line no-console
+        console.warn('Shape of left matrix are not equal to the shape of right matrix.');
```
Comment (maasencioh) [inline src/abstractMatrix.js]: I would rather throw an error

File: src/__tests__/matrix/utility.js [hunk 2]
```diff
@@ -212,6 +212,12 @@ describe('utility methods', () => {
     ]);
   });
 
+  it('emul', () => {
+    var matrix = new Matrix([[2, 4], [7, 1]]);
+    var matrix2 = new Matrix([[2, 1], [1, 1]]);
+    expect(matrix.emul(matrix2).to2DArray()).toStrictEqual([[4, 4], [7, 1]]);
+  });
+
```
Comment (maasencioh) [inline src/__tests__/matrix/utility.js]: I would add as well a test for non matching sizes

File: src/abstractMatrix.js (hunk 1 above)
Comment (Suckzoo) [reply src/abstractMatrix.js]: I just left warning messages like the other functions do.
I agree with your opinion. I'll change the code to throw an error :)

**Comment by Suckzoo** [comment]
@maasencioh  I'm done!

File: src/abstractMatrix.js [hunk 3]
```diff
@@ -980,6 +980,33 @@ export default function AbstractMatrix(superCtor) {
...
 
+    /**
+         * Returns the element-wise product between this and other
+         * @param {Matrix} other
+         * @return {Matrix}
+         */
+    emul(other) {
+      other = this.constructor.checkMatrix(other);
+      if (this.rows !== other.rows || this.columns !== other.columns) {
+        // eslint-disable-next-line no-console
```
Comment (targos) [inline src/abstractMatrix.js]: this line can be removed now

**Comment by targos** [comment]
Thanks for the contribution!
I just have a question regarding the name "emul". Is it common to call it like this? Maybe something more explicit would make the function easier to discover?

**Comment by Suckzoo** [comment]
@targos I just followed the conventions of the existing code. As `mmul` means matrix multiplication, I used `emul` as element-wise multiplication ¯\_(ツ)_/¯
Mathjs implements this function with the name `dotMultiply`. (reference: http://mathjs.org/docs/reference/functions.html) Choice is yours :)

**Comment by targos** [comment]
Hi, I'm sorry it took so long, but I just realized that this is already implemented. `mat1.mul(mat2)` does an element-wise multiplication (and you also have access to all other mathematical operators).
Thanks for the contribution anyway.

**Pull request closed without merging**